### PR TITLE
ProgressTableLevelIcon for progress table refactor

### PR DESCRIPTION
**Pull request by maureensturgeon**
This is a new component which represents a refactored portion of the [VirtualizedDetailView](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/sectionProgress/detail/VirtualizedDetailView.jsx) (see line 189). It is a combined effort between @cforkish and @maureensturgeon, and was extracted from the full refactor work in #38309.

The unit tests were PR'd previously: https://github.com/code-dot-org/code-dot-org/pull/38565

Level icon:
Screen Shot 2021-02-01 at 3 07 36 PM

You can refer to [this tech spec](https://docs.google.com/document/d/1SS1DYiLObiccRrVTo3LuLSe5OFiWla2qhsIWlfMvPIo/edit#) to see how this component fits into the full table architecture.

## Links

- [spec]()
- [jira](https://codedotorg.atlassian.net/browse/LP-1673)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx]: This whole return statement is a little difficult to read.  Maybe try extracting the anonymous function from L30-L50?  If it's still difficult to read, then maybe also extract L32-40 and L42-48?  (Extracting these parts gives you a place to write some comments which I think might be helpful, especially that the sublevels bit is to add space to get icons to line up (right?))

Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx]: I did a little refactor, let me know if you find it more readable!

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx]: if these are just helper functions to render parts of the component, they should be inside the class. for `sublevelSpace` i think that makes most sense since it doesn't really feel like a useful subcomponent to factor out.

this one though i think makes sense as a subcomponent, in which case this should become`LevelIcon(props)`, and line 58 would become `<LevelIcon icon={getIconForLevel(level, true)} /`>, and line 30 `<FontAwesome icon={props.icon} />`

**Review by cforkish — CHANGES_REQUESTED** [review]

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx]: (minor) Not sure what others think, but I like to name functions that return a React element `renderFoo()` so this function might be called `renderSublevelSpace()`.  Up to you.

**Review by jamescodeorg — COMMENTED** [review]
I think this is much clearer, thanks!  Will approve after changes suggested by Charlie.

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (cforkish) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx]: i support adopting this as a convention!

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx:22]: according to the react docs, function components are supposed to take a single "props" argument. I'm guessing the linter made you add those PropTypes in which case it must know this is a component, but can you confirm that React treats this as a component even with "props" implied?

**Review by cforkish — APPROVED** [review]
great refactor, this is much cleaner / more readable! 

i'd appreciate a confirmation about `LevelIcon` being a proper function component before you merge -- i'm guessing the "props" argument is implied by`{icon}` but would love to know for sure.

**Comment by maureensturgeon** [comment]
> great refactor, this is much cleaner / more readable!
> 
> i'd appreciate a confirmation about `LevelIcon` being a proper function component before you merge -- i'm guessing the "props" argument is implied by`{icon}` but would love to know for sure.

`{icon}` is just deconstructed props, it will work as a function component

File: apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx
Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableLevelIcon.jsx:22]: Oh - I saw this comment late! Yes, it is still treated as a component, `{icon}` is still a single "props" argument, just deconstructed

**Pull request merged**
